integration/kotlinx-coroutines-guava/src/ListenableFuture.kt

```diff
@@ -385,9 +385,24 @@ private class JobListenableFuture<T>(private val jobToCancel: Job): ListenableFu
         // this Future hasn't itself been successfully cancelled, the Future will return
         // isCancelled() == false. This is the only discovered way to reconcile the two different
         // cancellation contracts.
-        return auxFuture.isCancelled || (isDone && Uninterruptibles.getUninterruptibly(auxFuture) is Cancelled)
+        return auxFuture.isCancelled || auxFuture.completedWithCancellation
     }
 
+    /**
+     * Helper for [isCancelled] that takes into account that
+     * our auxiliary future can complete with [Cancelled] instance.
+     */
+    private val SettableFuture<*>.completedWithCancellation: Boolean
+        get() = isDone && try {
+            Uninterruptibles.getUninterruptibly(this) is Cancelled
+        } catch (e: CancellationException) {
+            true
+        } catch (t: Throwable) {
+            // In theory appart from CancellationException, getUninterruptibly can only
```

Comment:
Soundness matters at this level. It's probably a good idea to handle `ExecutionException` as it's expected, but throw from a `check` if any other exception type is thrown. `getUninterruptibly` is heavily load-bearing in Guava, so if it ever occurs that it doesn't follow its documented exception contract, we need to know.

I think there was a rethrow in an earlier version of the file that threw with a message about a defective Future implementation...

Reply:
Fair, narrowed it to `ExecutionException`.